src/compiler/factory.ts

```diff
@@ -324,17 +324,19 @@ namespace ts {
         const node = <ArrayLiteralExpression>createNode(SyntaxKind.ArrayLiteralExpression, location);
         node.elements = parenthesizeListElements(createNodeArray(elements));
         if (multiLine) {

```

Comment:
Setting it unnecessarily could introduce a new hidden class and cause deoptimizations, and increases the number of bytes per node.